### PR TITLE
Indirect Data Analysis: Suppress warning on missing optional output 

### LINKED ISSUE #20610: Cannot save or plot result in Indirect Elwin interface
### Expected behavior
After loading and running a nexus file in the Elwin tab of the Indirect Data Analysis GUI, Plot Result would be expected to plot the output data, and Save Result would be expected to save it to an external file.

### Actual behavior
Neither operation is successful - in both cases we get a 'workspace not found' error.

### Steps to reproduce the behavior
Go to `Interfaces->Indirect->Indirect Data Analysis` and, inside the Elwin tab, load a focused IRIS file (such as [this](https://github.com/mantidproject/mantid/files/1324565/irs26176_graphite002_red.zip)). Click Run, and then try to do Plot Result and Save Result.

### Platforms affected
All

### PR DESCRIPTION
**Pull request by martyngigg**
Description of work.

**To test:**

See the test instructions in the original issue. The missing `_elt` workspace should now not produce a warning. To produce the `_elt` workspace first use the Indirect Data Reduction interface to reduce a newer file such as `IRIS00073585` and save the result. Load this into the data analysis interface, click "Normalise to Lowest Temperature" and click run. The `_elt` workspace should now appear and plotting should show all 4 plots.

Fixes #20610 

**Release Notes** 

*Feature introduced this cycle. Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by NickDraper** [comment]
@brandonhewer Could you fit in a review for this PR soon please.

File: qt/scientific_interfaces/Indirect/IndirectTab.cpp [hunk 1]
```diff
@@ -620,20 +620,19 @@ QString IndirectTab::runPythonCode(QString code, bool no_output) {
...
-  } else {
+  if (warn && !workspaceExists) {
     const std::string plotSave = plotting ? "plotting" : "saving";
     const auto errorMessage = "Error while " + plotSave +
                               ":\nThe workspace \"" + workspaceName +
                               "\" could not be found.";
     const char *textMessage = errorMessage.c_str();
-    QMessageBox::warning(NULL, tr("Workspace not found"), tr(textMessage));
-    return false;
+    QMessageBox::warning(nullptr, tr("Indirect "), tr(textMessage));
```
Comment (samueljackson92) [inline qt/scientific_interfaces/Indirect/IndirectTab.cpp]: The bulk of this function is in a `if` statement. Consider flipping the condition and using an early exit?

**Review by samueljackson92 — CHANGES_REQUESTED** [review]
Tested this but still got a warning when trying to save a result, but not when trying to plot

File: qt/scientific_interfaces/Indirect/IndirectTab.cpp (hunk 1 above)
Comment (martyngigg) [reply qt/scientific_interfaces/Indirect/IndirectTab.cpp]: I'd need to go with:
```
  if(!warn) return workspaceExists;
  if(!workspaceExists) {
    const std::string plotSave = plotting ? "plotting" : "saving";
    const auto errorMessage = "Error while " + plotSave +
                              ":\nThe workspace \"" + workspaceName +
                              "\" could not be found.";
    const char *textMessage = errorMessage.c_str();
    QMessageBox::warning(nullptr, tr("Indirect "), tr(textMessage));
  }
  return workspaceExists;
```
which to me doesn't read as well. It's not performance critical code so the early exit isn't a huge advantage. What do you think?

Comment (samueljackson92) [reply qt/scientific_interfaces/Indirect/IndirectTab.cpp]: You're right. I don't think the condition can be easily flipped. Leave it as it is currently.

**Comment by martyngigg** [comment]
@samueljackson92 Thanks for catching that. I've pushed a fix.

**Review by samueljackson92 — APPROVED** [review]

**Comment by samueljackson92** [comment]
@martyngigg This now works for me. Wait for the builds to pass and this is good to go. 👍

**Pull request merged**